network/src/peer_manager/tests.rs

```diff
@@ -237,11 +236,11 @@ fn peer_disconnect_request() {
         .await;
     };
 
-    block_on(join(test, peer.start()));
+    join(test, peer.start()).await;
```

Comment:
does this mean we can't use `futures::executor::block_on` anymore if the `Future` contains some tokio-specific things?